COMRS-126: Enabling translation for the search by address field

rubailly opened the pull request.

rubailly committed:
COMRS-126: Enabling translation for the search by address field

tmvumbi2 commented on ui/app/registration/views/search.html:
```diff
@@ -26,8 +26,8 @@
                 <article class="fl" ng-if="::addressSearchConfig.show">
-                    <label for="name"><strong>{{::addressSearchConfig.label}}</strong></label>
-                    <input tabindex="6" type="text" id="addressFieldValue" ng-model="searchParameters.addressFieldValue" placeholder="{{::addressSearchConfig.placeholder}}">
+                    <label for="name"><strong>{{::'REGISTRATION_SEARCH_ADDRESS_LABEL_KEY' | translate}}</strong></label>
```
@rubailly the keys for the address search label and placeholder are configured in `registration/app.json` (under "patientSearch"). These key's values are available in the scope variables addressSearchConfig.label and addressSearchConfig.placeholder. We can therefore directly translate the scope variables (I.e. `{{::addressSearchConfig.label | translate }}`); this way we do not need to hard code the keys in the code (with the current solution, if the key changes in app.json, the translation will break). 

@ajinkyagadewar  this fix could benefit Bahmni as well, can we arrange to merge this back into Bahmni?

rubailly replied:
@tmvumbi2 - this has been completed

tmvumbi2 reviewed (changes requested):
@rubailly Internationalizing the label and placeholder for address search is a great addition :+1: The PR looks great, I left a single small feedback

rubailly committed:
COMRS-23: Removing hard-coded translation keys to use scope variables

tmvumbi2 committed:
COMRS-126 Translate the table's headers on search result

The table's header was displaying a key code (instead of a translated value) for the dynamically generated columns; this commit fixes the bug.
